docs: add first draft of LSP11 specs (#68)
lukso-network/LIPs: LSPs/LSP-11-BasicSocialRecovery.md

---
lip: 11
title: BasicSocialRecovery
author: 
discussions-to: https://discord.gg/E2rJPP4
status: Draft
type: LSP
created: 2021-2-14
requires: ERC725, LSP0, LSP2, LSP6
---

## Simple Summary
This standard describes a **basic social recovery** contract that can recover access to [ERC725](https://github.com/ERC725Alliance/ERC725/blob/develop/docs/ERC-725.md) contracts through the [LSP6-KeyManager](./LSP-6-KeyManager.md).

## Abstract

> This standard allows for recovering access to ERC725 contracts such as tokens, NFTs, and Universal Profiles by adding a new address to control them through the Key Manager after a recovery process.

Reviewer: ```suggestion
This standard provides a mechanism for recovering access to ERC725 contracts such as tokens, NFTs, and Universal Profiles by adding a new address to control them through the Key Manager after a recovery process.
```

> The social recovery contract should provide the most flexible and secure process, where the guardians, set originally by the owner, can vote to one address in each recoverProcessId. The address willing to recover can choose a recoverProcessId where he reached the guardiansThreshold and after successfully providing the right secret word, that produces the hash already set by the owner, he will be granted the owner permissions in the account recovered and a new hash will be set and all previous recoverProcessId will be invalidated.

Reviewer: ```suggestion
The social recovery contract should ensure a flexible and secure process where guardians, nominated by the owner, can vote for one address in each `recoverProcessId`. The address requesting recovery can choose a `recoverProcessId` where he reached the `guardiansThreshold` and after successfully providing the right secret word which produces the same hash as the secret word set by the owner, he will be granted the owner permissions in the recovered account and a new hash will be set and all previous `recoverProcessId` will be invalidated.
```

I tried to make this a bit more clear but please double check the content is still correct. 

I think this passage could do with a more clear bullet pointed process so its clear what happens step by step

## Motivation
Any Key could be lost or leaked due to a certain accident, so it's not advised to rely on one singular key to control ERC725 contracts through the Key Manager and a social recovery contract is needed in this case.

In the case above, the user can simply reach out to his guardians and ask them to vote for a certain address. 
There are many possible choices for whom to select as a guardian. The three most common choices are:

> - Other devices (or paper mnemonics) owned by the wallet holder themselves

Reviewer: ```suggestion
- Other EOAs controlled by the wallet holder themselves (via paper mnemonics or cold storage wallets) 
```

- Friends and family members (EOAs or Universal Profiles)

> - Institutions, which would vote for the address provided if they get a confirmation of your phone number or email or perhaps in high-value cases to verify you personally by video call

Reviewer: ```suggestion
- Institutions, which could vote for a provided address if they get a valid confirmation via phone number, email or video call
```

```

Recover the linked account by setting in All Permissions (combined) for the msg.sender after it reached the recoverThreshold and given the right plainSecret that produce the `secretHash`.

_Parameters:_

- `recoverProcessId`: the recover process id in which the `msg.sender` should have reached the threshold.

- `plainSecret`: the plain secret that should produce the `secretHash` with _keccak256_ function.

- `newHash`: the new secret hash to set.

### Setup

In order to allow the social recovery contract to recover the linked account and add new permissions, this contract should have `ADDPERMISSIONS` and `CHANGEPERMISSIONS` set inside the **linked account** under this ERC725Y Data Key.

```json
{
    "name": "AddressPermissions:Permissions:<address>",
    "key": "0x4b80742de2bf82acb3630000<address>",
    "keyType": "MappingWithGrouping",
    "valueType": "bytes32",
    "valueContent": "BitArray"
}
```

### Events

## Rationale

This standard was inspired by the current recovery process in some crypto wallets but this recovery process is a balance between a secret hash and guardians.

> In this case, you can ensure that you're guardians can't act maliciously and would need a secret word to recover. The same goes for the secret word if it's exposed, only addresses who reached the guardiansThreshold can recover using it.

Reviewer: ```suggestion
In this case, it is ensured that guardians can't act maliciously and would need a secret word to recover. The same goes for the secret word if it is exposed, only addresses who reached the guardiansThreshold can use it to recover an account.
```

---
lip: 11
title: BasicSocialRecovery
author: 
discussions-to: https://discord.gg/E2rJPP4
status: Draft
type: LSP
created: 2021-2-14
requires: ERC165, ERC725, LSP2, LSP6
---

## Simple Summary

> This standard describes a **basic social recovery** contract that can recover access to [ERC725] contracts through the [LSP6-KeyManager](./LSP-6-KeyManager.md).

Reviewer 2: ```suggestion
This standard describes a **basic social recovery** contract that can recover access to [ERC725] contracts through a [LSP6-KeyManager](./LSP-6-KeyManager.md).
```

## Abstract

This standard provides a mechanism for recovering access to ERC725 contracts such as tokens, NFTs, and Universal Profiles by adding a new controller address through the Key Manager after a recovery process.

The social recovery contract should ensure a flexible and secure process where guardians, nominated by the owner, can select one address. The address after reaching the `guardiansThreshold` and after successfully providing the secret word which produces the same hash as the secret hash set by the owner, will be granted the owner permissions in the recovered target and a new hash will be set and a new recovery counter will be created.

## Motivation

> Any Key could be lost or leaked due to a certain accident, so it's not advised to rely on one singular key to control ERC725 contracts through the Key Manager and a social recovery contract is needed in this case.

Reviewer 2: ```suggestion
Any Key could be lost or leaked due to a certain accident, so it is not advised to rely on a singular key to control ERC725 contracts through the Key Manager and a social recovery contract is needed in this case.
```

> In the case above, the user can simply reach out to his guardians and ask them to vote for a certain address. 

Reviewer 2: ```suggestion
In the case above, the user can reach out to his guardians and ask them to vote for a specific address. 
```

> There are many possible choices for whom to select as a guardian. The three most common choices are:

Reviewer 2: ```suggestion
There are many possible options for whom to select as a guardian. The three most common options are:
```
